### ISSUE TITLE
Talk: Deep Magic of JS performance

### ISSUE THREAD

**Issue by andyhall** [tokyojs/tokyojs.github.io#5]
## Deep Magic of JS performance (for v8/Chrome/Node)

While tuning a JS game, I recently spent a bunch of time learning about v8 internals and how they affect performance, and this talk is to pass on the lessons I learned. Focusing on v8 only because it's the easiest to get information about, but most advice should apply to other engines.

* What the optimizing compiler does, what makes it fail, how to detect when it's failed
* How the engine infers object types, and how they affect performance
* How to measure performance reliably

Basically I will avoid trivia, like "make your for loops count down instead of up! (*)", and instead try to give people an intuitive sense of what kind of code modern JS engines can and can't optimize. 

(* This is wrong, by the way!)

## Details:
Name: Andy Hall
Duration: 45 minutes-ish

**Comment by brj** [comment]
> (* This is wrong, by the way!)

Really? I've been wrong for so long 😭

**Comment by andyhall** [comment]
>  I've been wrong for so long :sob: 

Haha, just saw this. Well, let's say it *should* be wrong, if V8 is doing its job! :grinning: 

But in practice it depends on some weird internals, which is what I'll try to explain!

**Comment by bucaran** [comment]
BTW the meetup has been scheduled for the 8/12 (Friday). Can you confirm your attendance?

**Comment by andyhall** [comment]
Yep! I confirmed on meetup.

**Comment by bucaran** [comment]
I mean, your talk. Would you like to fill the 3rd slot (20:00)?

**Comment by andyhall** [comment]
Sure, I'd love to!

**Comment by bucaran** [comment]
Yay!
